include/revng/Model/Type.h

```diff
@@ -81,6 +81,10 @@ class Identifier : public llvm::SmallString<16> {
   bool verify(VerifyHelper &VH) const;
 };
 
+Identifier makeValidIdentifier(const Identifier &);
```

Comment:
I'd like this to be a `static` factory method within Identifier from a `StringRef`

Reply:
You have already done that actually. It wasn't there in the branch I rebased this patchset from, so I didn't notice that. I'll drop these helpers and fix your factory, which suffers of some of the same shortcomings you have pointed out about my helpers.

Reply:
Done, take a look and let me know what you think.